pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceConsumer.java

```diff
@@ -406,34 +497,90 @@ public static void main(String[] args) throws Exception {
+                                + "--- {}  Txn/s --- AckRate: {} msg/s"
+                                + "--- Latency: mean: {} ms - med: {} "
+                                + "- 95pct: {} - 99pct: {} - 99.9pct: {} - 99.99pct: {} - Max: {}",
+                        intFormat.format(total),
+                        dec.format(rate), dec.format(throughput),
+                        totalTxnOpSuccessNum,
+                        totalTxnOpFailNum,
+                        dec.format(rateOpenTxn),
+                        dec.format(rateAck),
+                        dec.format(reportHistogram.getMean()),
+                        reportHistogram.getValueAtPercentile(50), reportHistogram.getValueAtPercentile(95),
+                        reportHistogram.getValueAtPercentile(99), reportHistogram.getValueAtPercentile(99.9),
+                        reportHistogram.getValueAtPercentile(99.99), reportHistogram.getMaxValue());
+            }else {
             log.info(
-                    "Throughput received: {} msg --- {}  msg/s -- {} Mbit/s --- Latency: mean: {} ms - med: {} - 95pct: {} - 99pct: {} - 99.9pct: {} - 99.99pct: {} - Max: {}",
+                    "Throughput received: {} msg --- {}  msg/s -- {} Mbit/s  "
+                            + "--- Latency: mean: {} ms - med: {} "
+                            + "- 95pct: {} - 99pct: {} - 99.9pct: {} - 99.99pct: {} - Max: {}",
                     intFormat.format(total),
                     dec.format(rate), dec.format(throughput), dec.format(reportHistogram.getMean()),
                     reportHistogram.getValueAtPercentile(50), reportHistogram.getValueAtPercentile(95),
                     reportHistogram.getValueAtPercentile(99), reportHistogram.getValueAtPercentile(99.9),
                     reportHistogram.getValueAtPercentile(99.99), reportHistogram.getMaxValue());
-
             reportHistogram.reset();
             oldTime = now;
-        }
+        }}
 
         pulsarClient.close();
     }
 
-    private static void printAggregatedThroughput(long start) {
+    private static void printAggregatedThroughput(long start, Arguments arguments) {
         double elapsed = (System.nanoTime() - start) / 1e9;
         double rate = totalMessagesReceived.sum() / elapsed;
         double throughput = totalBytesReceived.sum() / elapsed * 8 / 1024 / 1024;
+        long totalEndTxnSuccess = 0;
+        long totalEndTxnFail = 0;
+        long totalnumMessageAckFailed = 0;
+        double rateAck = totalMessageAck.sum() / elapsed;
+        double rateOpenTxn = 0;
+        if (arguments.isEnableTransaction) {
+            totalEndTxnSuccess = totalEndTxnOpSuccessNum.sum();
+            totalEndTxnFail = totalEndTxnOpFailNum.sum();
+            rateOpenTxn = (totalEndTxnSuccess + totalEndTxnFail) / elapsed;
+            totalnumMessageAckFailed = totalMessageAckFailed.sum();
+        }
+        if(arguments.isEnableTransaction){
```

Comment:
I still suggest that we should not modify the logic of the original log, and only print the relevant parameters separately when enabling transaction.

Reply:
I kept the original log logic in else.Isn't it done this way?

Reply:
I think we only need to print the log added by transaction when enable, don't need change the original log logic.

Reply:
Is there any logic to change the original log?